hoomd/write/gsd.py

```diff
@@ -167,7 +174,18 @@ def __init__(self,
                           dynamic=[dynamic_validation],
                           _defaults=dict(filter=filter, dynamic=dynamic)))
 
-        self._log = None if log is None else _GSDLogWriter(log)
+        if all((logger is not None, log is not None)):
+            warnings.warn(
+                f"log and logger keyword arguments passed to {self}."
+                f" Keyword argument \"log\" is deprecated since v3.9.0."
+                f" Ignoring log and using logger instead.", DeprecationWarning)
```

Comment:
@b-butler has many times advocated for the use of `FutureWarning`. Whichever we choose, we should be consistent here and in #1485.